### ISSUE TITLE
Callback needs to run 2-3 times before it changes the parameters value (e.g. devices.ecomax.boiler_control).

### ISSUE THREAD

**Issue by VespuCore** [denpamusic/PyPlumIO#3]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### I'm having the following issue:

When I try to use the following example (in [README.md](https://github.com/denpamusic/PyPlumIO#writing)) to write a value to a parameter (e.g. **boiler_control**), it doesn't write the value on the first try, I need to wait for the callback to run 2-3 times before it changes the actual value.

```py
from pyplumio import TcpConnection
from pyplumio.devices import DeviceCollection

async def my_callback(devices: DeviceCollection, connection):
    if devices.has("ecomax") and devices.ecomax.has("boiler_control"):
    	devices.ecomax.boiler_control = 1
        connection.close()

connection = TcpConnection(host=myipaddress, port=myport)
connection.run(main, interval=3)
```

So I need to wait 6-9 seconds (2-3x interval) for the code to write the value, and that means I can't use `connection.close()`, since it closes the connection after the first time of the callback being run.

When the script contains `connection.close()`, I get no output and nothing gets changed.

![image](https://user-images.githubusercontent.com/53978081/175300144-511c4cc5-ba25-4468-bb8b-f38b16dcc9b3.png)

But when removing `connection.close()` from the script, I waited 4 seconds and killed the script, but the value was changed.

![image](https://user-images.githubusercontent.com/53978081/175300394-b2d2808f-60be-457e-84f5-da0d1f1a8a49.png)

When I did `print(devices.ecomax.data)` the product model printed was `ecoMAX920P1-T` instead of `EcoMax 860P`.

The home assistant plugin works perfectly, unlike the python script.

### My environment is:

```markdown
- OS: Debian 11.3 (Latest Raspberry Pi x64 image)
- OS: Windows 10
- Python: 3.10
- Device: EcoMax 860P
- PyPlumIO Version: 0.1.17 (Latest)
```

### I have following devices connected:

- [ ] ecoMAX 3xx series
- [X] ecoMAX 8xx series
- [ ] ecoMAX 9xx series
- [ ] Expansion module B
- [ ] Expansion module C
- [ ] ecoSTER 200/ecoSTER Touch
- [ ] ecoLAMBDA
- [ ] ecoNET 300

### I'm connecting to my devices using:

Ethernet/WiFi to RS-485 converter

### I'm seeing following log messages:

```shell
Frame error: Incorrect frame length. Expected 367 bytes, got 356 bytes
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by VespuCore** [comment]
`Frame error: Incorrect frame length. Expected 367 bytes, got 356 bytes` appears sometimes, not always.

**Comment by denpamusic** [comment]
Hi, thank you for the detailed feedback!

This happens because request to change the parameter doesn't get written immediately, but gets added to the write queue instead. In current version PyPlumIO doesn't check if write queue has been fully processed before closing the connection. This doesn't happen in hass, since hass doesn't close connection unless integration is deactivated or hass is shut down, at which point all requests to set parameters are probably already processed anyway.

I'm currently developing branch 0.2.x that does ensure that every request was processed by using native asyncio queues. It'll also include a lot of other changes, such as ability to add a callback on value change. This branch is still far from being ready though, so I'll try to implement a simple check and include it in the last release of 0.1.x branch.

Incorrect frame length on the other hand is just a warning, that can be safely ignored. In fact it will be demoted to debug message in 0.2.x, since it happens a lot when using network converters and by itself pretty harmless - it's just means, that library was not able to capture full frame, either due to interference on rs485 wires, connection hiccup or just slow hardware.

**Comment by VespuCore** [comment]
Hi, thank you for the great work, can't wait for **0.2.x**!

Anyway, should I close this issue?

**Comment by denpamusic** [comment]
You’re welcome! First release of 0.2.x will probably be ready by the end of next week, hope you’ll like it :) As for this issue, please leave it open, as I intend to implement fix for this in 0.1.x.

**Comment by denpamusic** [comment]
First version of [0.2.x](https://github.com/denpamusic/PyPlumIO/releases/tag/v0.2.0) branch has just been released. It features totally new architecture, that considerably speeds up queue processing and by extent fixes this issue. After careful consideration I decided against backporting these changes to 0.1.x due to amount work it requires and little benefit of it. Since user-base of this project is still so small, I should be able to get away with it😁

HASS integration will be ported to this branch within next three to four days.

As for your example, you now have a couple of options to change boiler control parameter using new syntax. With any of these examples connection will not be closed until boiler_control parameter has been sent to the device. 

1. using `Device.set_value(name, value)` method. "on" here is an alias for 1:
```python
import asyncio
import pyplumio

async def main():
  async with pyplumio.open_tcp_connection(myipaddress, myport) as connection:
    ecomax = await connection.get_device("ecomax")
    await ecomax.set_value("boiler_control", "on")
    
asyncio.run(main())
```
2. getting parameter instance from the device and using `Parameter.turn_on()` method:
```python
import asyncio
import pyplumio

async def main():
  async with pyplumio.open_tcp_connection(myipaddress, myport) as connection:
    ecomax = await connection.get_device("ecomax")
    boiler = await ecomax.get_parameter("boiler_control")
    boiler.turn_on()
    
asyncio.run(main())
```
